src/style-spec/function/definitions/match.js

```diff
@@ -107,15 +107,12 @@ class Match implements Expression {
         for (let i = 0; i < labels.length; i++) {
             if (i > 0) lookup += ', ';
             const label = labels[i];
-            lookup += `${String(label)}: ${outputs[this.cases[label]]}`;
+            lookup += `${JSON.stringify(label)}: ${outputs[this.cases[label]]}`;
         }
 
         const lookupObject = ctx.addVariable(`{${lookup}}`);
 
-        const inputType = this.inputType.kind !== 'Array' ?
-            `$this.types.${this.inputType.kind}` : JSON.stringify(this.inputType);
-
-        return `(${lookupObject}[$this.as(${input}, ${inputType})] || ${ctx.addExpression(this.otherwise.compile(ctx))})();`;
```

Comment:
What was the purpose of using `$this.as` here before?

Reply:
I believe it was always redundant, since type checking should already ensure that the input value is the right type.